proposals/evaluation/uc-2.5.2-client-constraints.md

```diff
@@ -0,0 +1,31 @@
+# Limiting application access while not acting as resource controller
+
+https://solid.github.io/authorization-panel/authorization-ucr/#uc-client-constraints
+
+### ACP
+
+ACP allows combining multiple matchers with `acp:allOf`, this allows expressin policy which uses `acp:agent` and `acp:client` in combined matchers.
+
+```ttl
+PREFIX acp: <http://www.w3.org/ns/solid/acp#>
+PREFIX acl: <http://www.w3.org/ns/auth/acl#>
+PREFIX ex: <https://example.com/>
+
+ex:accessControlResourceA
+  acp:resource ex:resourceX ;
+  acp:accessControl ex:accessControlB .
+
+ex:accessControlB acp:apply ex:PolicyC .
+
+ex:PolicyC
+  acp:allOf ex:agentMatcherD, ex:clientMatcherD ;
+  acp:allow acl:Read .
+
+ex:agentMatcherD
+  acp:agent ex:Bob .
+
+ex:clientMatcherD
+  acp:client ex:Projectron .
```

Comment:
Note that this is equivalent effectively to:

```ttl
ex:PolicyC
  acp:allOf ex:matcherD ;
  acp:allow acl:Read .

ex:matcherD
  acp:agent ex:Bob ;
  acp:client ex:Projectron .
```

Which is interesting because we could imagine a future where a matcher dedicated to agent Bob would be maintained by them and they might edit the clients they trust in and out.

For example something such as:

```ttl
ex:ReaderAgentPolicy
  acp:anyOf ex:bobMatcher, ex:aliceMatcher ;
  acp:allow acl:Read .

ex:bobMatcher
  acp:agent ex:Bob ;
  acp:client ex:Projectron .

ex:aliceMatcher
  acp:agent ex:Alice ;
  acp:client ex:Projectron, ex:OtherApp .
```

Obviously, that would be if you trust Alice and Bob to delegate read access. But you could also lock in the fact that Bob has read access and let him handle the clients he trusts via a clientMatcher (and have 1 policy per agent, ex:BobReaderAgentPolicy)...